common/git-common/src/main/java/org/arquillian/pact/common/git/GitOperations.java

```diff
@@ -81,18 +83,31 @@ public Ref checkoutTag(Git git, String tag) {
     public boolean isLocalBranch(final Git git, final String branch) {
```

Comment:
no in current approach because do a push implies you have a git object. If you add git object as attribute of GitOperations, nothing prevents you to call push method without having any Git object initialized.